contracts/BC_fusion/StakePool.sol

```diff
@@ -0,0 +1,266 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts-upgradeable/utils/structs/DoubleEndedQueueUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/ERC20PermitUpgradeable.sol";
+
+import "./System.sol";
+
+interface IStakeHub {
+    function unbondPeriod() external view returns (uint256);
+    function transferGasLimit() external view returns (uint256);
+}
+
+contract StakePool is Initializable, ReentrancyGuard, System, ERC20PermitUpgradeable {
+    using DoubleEndedQueueUpgradeable for DoubleEndedQueueUpgradeable.Bytes32Deque;
+
+    /*----------------- constant -----------------*/
+    uint256 public constant MAX_CLAIM_NUMBER = 20;
+
+    string public constant EIP712_NAME = "BSC validator pool";
+    string public constant EIP712_VERSION = "1.0.0";
+    string public constant ERC20_NAME = "BSC staked BNB";
+    string public constant ERC20_SYMBOL = "stBNB";
+
+    /*----------------- storage -----------------*/
+    address public validator;
+
+    // for slash
+    bool private _freeze;
+    uint256 private _remainingSlashBnbAmount;
+
+    uint256 private _totalPooledBNB; // total reward plus total BNB staked in the pool
+
+    // hash of the unbond request => unbond request
+    mapping(bytes32 => UnbondRequest) private _unbondRequests;
+    // user => unbond request queue(hash of the request)
+    mapping(address => DoubleEndedQueueUpgradeable.Bytes32Deque) private _unbondRequestsQueue;
+    // user => locked shares
+    mapping(address => uint256) private _lockedShares;
+    // user => personal unbond sequence
+    mapping(address => uint256) private _unbondSequence;
+
+    struct UnbondRequest {
+        uint256 sharesAmount;
+        uint256 bnbAmount;
+        uint256 unlockTime;
+    }
+
+    /*----------------- events -----------------*/
+    event Delegated(address indexed sender, uint256 sharesAmount, uint256 bnbAmount);
+    event RewardReceived(uint256 bnbAmount);
+    event UnbondRequested(address indexed sender, uint256 sharesAmount, uint256 bnbAmount, uint256 unlockTime);
+    event UnbondClaimed(address indexed sender, uint256 sharesAmount, uint256 bnbAmount);
+
+    /*----------------- modifiers -----------------*/
+
+    /*----------------- external functions -----------------*/
+    function initialize(address _validator, uint256 minSelfDelegationBNB) public payable initializer {
+        __EIP712_init_unchained(EIP712_NAME, EIP712_VERSION);
+        __ERC20_init_unchained(ERC20_NAME, ERC20_SYMBOL);
+
+        validator = _validator;
+
+        assert(msg.value != 0);
+        _bootstrapInitialHolder(msg.value);
+
+        // transfer to stakeHub
+        _transfer(validator, STAKE_HUB_ADDR, minSelfDelegationBNB);
```

Comment:
what you transfer here is stBNB, how can it calll as `minSelfDelegationBNB`?

Reply:
For the first mint process, amount(stBNB) === amount(BNB)